### PR TITLE
Make interpolation order selectable and set linear as default

### LINKED ISSUE #131: Tabular dispersions might break Solver2x2
Using something like this [Si_Palik.txt](https://github.com/PyEllips/pyElli/files/11241834/Si_Palik.txt) to generate a dispersion leads to rounding errors, where k varies around 0 in the IR and can become negative. 

This breaks Solver2x2 as described in #130.

```
lbda = np.linspace(200, 1700, 10)
table = elli.Table(Si_Palik.index.values, Si_Palik["n"] + 1j * Si_Palik["k"])
table.get_refractive_index(lbda)

array([0.9151964 +2.75630140e+00j, 6.68599419+2.47306141e+00j,
       4.14435409+4.37175170e-02j, 3.78316134+1.26652976e-02j,
       3.64459034+2.26450024e-02j, 3.56345879+1.10329426e-02j,
       3.5193    -7.62329653e-21j, 3.50295359-2.04540250e-06j,
       3.47835012+3.04455261e-08j, 3.46418165+7.81489545e-09j])
```

Edit: I think the cubic interpolation leads to oscillations in this case. Setting it to linear fixes the issue, but i have no idea how we should handle this correctly.

### PR DESCRIPTION
**Pull request by MarJMue**
fixes #131

### CODE REVIEW

File: src/elli/dispersions/table_epsilon.py [hunk 1]
```diff
@@ -25,7 +27,7 @@ class TableEpsilon(Dispersion):
         The interpolation in the given wavelength range.
     """
 
-    single_params_template = {"lbda": None, "epsilon": None}
+    single_params_template = {"lbda": None, "epsilon": None, "kind": "linear"}
```
Comment (domna) [inline src/elli/dispersions/table_epsilon.py]: I wouldn't insert this via the single_params as this is not really a fitting parameter and we may use this dict to update the fitting parameters at some point (which is not really relevant for the table but I think we should still keep it consistent).

I would rather add it to the `__init__` directly, i.e., adding something like
```
self.kind = kwargs.get('kind', 'linear')
```
in the `__init__` or directly as a parameter.

File: src/elli/dispersions/table_index.py [hunk 2]
```diff
@@ -26,7 +28,7 @@ class Table(IndexDispersion):
         The interpolation in the given wavelength range.
     """
 
-    single_params_template = {"lbda": None, "n": None}
+    single_params_template = {"lbda": None, "n": None, "kind": "linear"}
```
Comment (domna) [inline src/elli/dispersions/table_index.py]: Here applies the same.

File: src/elli/dispersions/table_epsilon.py (hunk 1 above)
Comment (MarJMue) [reply src/elli/dispersions/table_epsilon.py]: ok good. i was not completely sure if all other kwargs get propagated or are ignored.

File: src/elli/dispersions/table_epsilon.py [hunk 3]
```diff
@@ -29,6 +31,8 @@ class TableEpsilon(Dispersion):
     rep_params_template: Dict[str, Any] = {}
 
     def __init__(self, *args, **kwargs) -> None:
+        self.kind = kwargs.pop("kind", "linear")
```
Comment (domna) [inline src/elli/dispersions/table_epsilon.py:34]: ah yes, right. Has to be poped

**Review by domna — APPROVED** [review]

**Comment by domna** [comment]
Just one note. I think it's good to go but wouldn't fix #131 entirely, because using a user specified cubic would still fail the 2x2 solver. I think, however, it's fine to close and deal with the issue in #130.

**Comment by MarJMue** [comment]
Defaults are important and with this it becomes unlikely, that some RII dispersion provokes the error from the fix of #130.

**Pull request merged**